workspaces/marketplace/plugins/marketplace/src/components/MarketplacePluginInstallContent.tsx

```diff
@@ -36,6 +35,86 @@ import {
   CodeEditor,
   useCodeEditor,
 } from './CodeEditor';
+import Box from '@mui/material/Box';
+import Grid from '@mui/material/Grid';
+import Card from '@mui/material/Card';
+import CardContent from '@mui/material/CardContent';
+import CardHeader from '@mui/material/CardHeader';
+import FileDownloadOutlinedIcon from '@mui/icons-material/FileDownloadOutlined';
+import Tabs from '@mui/material/Tabs';
+import Tab from '@mui/material/Tab';
+import FormGroup from '@mui/material/FormGroup';
+import FormControlLabel from '@mui/material/FormControlLabel';
+import Checkbox from '@mui/material/Checkbox';
+import Button from '@mui/material/Button';
+import Typography from '@mui/material/Typography';
+import ReactMarkdown from 'react-markdown';
+import { useNavigate } from 'react-router-dom';
+import { useTheme } from '@mui/material/styles';
+
+const generateCheckboxList = (packages: MarketplacePackage[]) => {
+  const hasFrontend = packages.some(
+    pkg => pkg.spec?.backstage?.role === 'frontend-plugin',
+  );
+  const hasBackend = packages.some(
+    pkg => pkg.spec?.backstage?.role === 'backend-plugin',
+  );
+
+  const checkboxes = [
+    { label: 'Install front-end plugin', show: hasFrontend },
+    { label: 'Install back-end plugin', show: hasBackend },
+    { label: 'Install software templates', show: true }, // TODO, now always show
+  ];
+
+  return checkboxes.filter(cb => cb.show);
+};
+
+const CheckboxList = ({ packages }: { packages: MarketplacePackage[] }) => {
+  const checkboxes = generateCheckboxList(packages);
+  const [checked, setChecked] = useState<{ [key: string]: boolean }>({});
+
+  const handleChange = (label: string) => {
+    setChecked(prev => ({ ...prev, [label]: !prev[label] }));
+  };
+
+  return (
+    <FormGroup>
+      {checkboxes.map((cb, index) => (
+        <FormControlLabel
+          key={index}
+          control={
+            <Checkbox
+              checked={checked[cb.label] ?? false}
+              onChange={() => handleChange(cb.label)}
+            />
+          }
+          label={cb.label}
+        />
+      ))}
+    </FormGroup>
+  );
+};
+
+interface TabPanelProps {
+  children: string;
+  index: number;
+  value: number;
+}
+
+const TabPanel = ({ children, index, value }: TabPanelProps) => {
+  if (value !== index) return null;
+
+  return (
+    <Box
+      role="tabpanel"
+      sx={{ flex: 1, overflow: 'auto', p: 2, scrollbarWidth: 'thin' }}
+    >
+      <Typography component="div">
+        <ReactMarkdown>{children}</ReactMarkdown>
+      </Typography>
```

Comment:
Please reuse the Markdown component:

```suggestion
      <Markdown content={...} />
```